crates/task-impls/src/consensus.rs

```diff
@@ -1111,11 +1046,31 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
 
                 // Add to the storage that we have received the VID disperse for a specific view
                 self.vid_shares.insert(view, disperse);
-                self.vote_if_able(&event_stream).await;
+                if self.vote_if_able(&event_stream).await {
+                    self.current_proposal = None;
+                }
             }
             HotShotEvent::ViewChange(new_view) => {
                 debug!("View Change event for view {} in consensus task", *new_view);
 
+                if *new_view == 0 {
+                    // Start polling for proposals for the first view
+                    self.quorum_network
+                        .inject_consensus_info(ConsensusIntentEvent::PollForProposal(1))
+                        .await;
+
+                    self.quorum_network
+                        .inject_consensus_info(ConsensusIntentEvent::PollForDAC(1))
+                        .await;
+
+                    if self.quorum_membership.get_leader(TYPES::Time::new(1)) == self.public_key {
+                        debug!("Polling for quorum votes for view {}", *self.cur_view);
+                        self.quorum_network
+                            .inject_consensus_info(ConsensusIntentEvent::PollForVotes(0))
+                            .await;
+                    }
+                }
+
```

Comment:
I don't think this really matters either way, but would it make sense to move these to the `ConsensusTaskState::create_from` function in `hotshot/src/tasks/task_state.rs`?

Reply:
yeah it would make sense I can move it there

Reply:
There was aleardy a function for this so I added it there